### DISCUSSION TITLE
Navigation buttons not working any more after pausing

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#14] [Q&A] [answered]
Today I had again this hanging in the app. It looks like this

![Screenshot_20230107-152712_Cruiser](https://user-images.githubusercontent.com/121371629/211160021-746b23c2-6f3a-4901-b7fe-cf1ef3dbe4fc.jpg)
You can press which ever button you want, no one is working. The more you push on the button the colors of the left 3 Buttons change in: sometimes all 3 are blue sometimes only one in the middle sometimes the two at the edge are blue. The only thing that you can do is using the back key to close the app. When you start it again, all works normal.

_Originally posted by @DolomitesTom in https://github.com/devemux86/cruiser/discussions/7#discussioncomment-4620601_

**Comment by DolomitesTom** [comment]
This bug is also available in Kurviger 2. It happens when you pause navigation and then open another app, for example for taking a photo and then want to resume the navigation. The buttons at the left are not working anymore.

**Comment by devemux86** [comment]
Are the right buttons working (e.g. zoom) at that moment?
Or the map scroll / zoom with gestures?

**Comment by devemux86** [comment]
Also, have you enabled any of the following at the time?
- Menu Tools | Screen lock
- Settings | Navigation | Screen saver

**Comment by DolomitesTom** [comment]
> Are the right buttons working (e.g. zoom) at that moment?
> Or the map scroll / zoom with gestures?

yes, these are working (when I remind correctly)

> have you enabled any of the following at the time?
>     Menu Tools | Screen lock
>     Settings | Navigation | Screen saver

no, for sure not.

**Comment by devemux86** [comment]
@DolomitesTom 
_I moved it to a discussion to have all conversations in one place._

**Reply by DolomitesTom** [reply]
Ok

**Comment by DolomitesTom** [comment]
Did you already have a chance to check what could cause this problem? As told I don't use screen save, but open another app, for example google maps to search a location and when coming back, this problem occurs (sometimes, not always). Other example is. I close the external overlay app for traffic warnings, then pause the navigation, maybe call the app launcher, then switch off the display to save power, after a while (half an hour for a drink or something) want to resume navigation and then I have the effect that none of the left buttons is working any more. Workaround is always quit cruiser and start it again with automatic reloading of the last route.

**Comment by devemux86** [comment]
> Did you already have a chance to check what could cause this problem?

I'm trying to repeat your examples, but so far unfortunately I can't reproduce it.

Does it happen on other phones?

**Comment by devemux86** [comment]
Are the buttons on auto-hide?
What happens when you disable auto-hide?

**Comment by DolomitesTom** [comment]
> Are the buttons on auto-hide?

yes, always

> What happens when you disable auto-hide?

not tested yet, need to try next tour (which can take some days due to starting snow period... hopefully not too long... cold is not a problem, ice and snow on the road is... need to think about spikes on the tires)

**Comment by devemux86** [comment]
Please try - if possible - also on _another phone_.
(to see it is a device or apps combination case)

> Are the buttons on auto-hide?
> yes, always

Interesting, please also try without auto hide to see what happens.

Are you sure that only the 3 buttons on the left show this behavior,
BUT not the 3 buttons on the right or the compass?

And can the app be used during this?
Can you scroll / zoom / long press the map?

> need to try next tour

Can't you reproduce it just starting a navigation at home
and switching apps?

**Comment by DolomitesTom** [comment]
> Please try - if possible - also on another phone.

can try this on the next tour, can also switch off auto-hide next time (even if this burns the buttons into the OLED screen, therefore will only do this on my main phone which is already full of it :-) )

> Can't you reproduce it just starting a navigation at home
> and switching apps?

not too easy, it doesn't happen all the time, only suddenly, but regularily on almost every daytrip. I have just tried it and had an additional strange effect. **All the left buttons didn't appear again** when pushing on the map, but only the right 3. All 3 worked, the compass not, no reaction while pressing on the compass. After a while the left 3 appeared again and I could resume the tour, so no crash, but somehow a not expected behaviour. I could provide the screenshot via PN or email due to privacy concerns, don't want to push it here, is this possible ?
While writing I could reproduce this effect also on my second phone: left 3 buttons didn't appear again, only after around one minute more they appeared again and even I could press on resume tour

> Are you sure that only the 3 buttons on the left show this behavior,
> BUT not the 3 buttons on the right or the compass?

Yes absolutely sure... the compass also showed no reaction, but the plus/minus and GPS positioning was working

> And can the app be used during this?
> Can you scroll / zoom / long press the map?

partly, scroll, zoom yes, longpress the map didn't try when this happened with the 3 blue buttons on the left: when this happens, nothing helps any more, not even waiting, only quit the app and start it again.

Update: just had again the effect that the 3 buttons on the left side didn't appear again... long pressing on the map has no effect: what should happen when long pressing on the map?

**Comment by devemux86** [comment]
> I could provide the screenshot via PN or email due to privacy concerns, don't want to push it here, is this possible ?

You can send me an email (GitHub profile at gmail).

> While writing I could reproduce this effect also on my second phone

This is important, so it is not an isolated case.

Did you do it at home? How, what steps?

> what should happen when long pressing on the map?

Currently long press on map inside navigation is not allowed (e.g. change waypoints).
But it will be available again in the next Cruiser update.

Thanks very much for the extra details.
I think now I have a clearer picture and can look further into what is going on.

---

Can you send me also the app settings file?
(Settings | Application | Preferences | Export)

To see if the auto hide is causing the problems,
please make next tests (even in home) having disabled all auto hide settings:
- Settings | Location | Hide compass
- Settings | Map | Hide map scale
- Settings | Application | Hide screen buttons

**Reply by DolomitesTom** [reply]
Thanks for your great support, email is on the way

    Settings | Location | Hide compass --> active
    Settings | Map | Hide map scale --> active
    Settings | Application | Hide screen buttons  --> active by default (now off for testing)

**Comment by DolomitesTom** [comment]
apropos, I said the described error was already in K2. Today I had it again while planning a route. No overlay apps were active. Just put K2 repeatedly in the background, open google maps to search for some locations which I want to insert into the route and suddenly the left tree buttons again where not working.
The right buttons where still working okay. Even the compass has worked, but nothing else. I had to quit the app.

![Screenshot_20230120-145906_Kurviger](https://user-images.githubusercontent.com/121371629/213779554-3e254001-fac0-47c4-bb10-87d568040d74.jpg)

**Reply by devemux86** [reply]
Thanks for the extra info.

Sorry for asking again, but I need to be sure of a few things:
- It happened on a 2nd device too?
- Please have you tried with **disabled** the auto hide buttons?
This is _very important_ to see if the auto hide is the cause.

**Reply by DolomitesTom** [reply]
No, this was only on my first phone (Galaxy S8+). There the auto hide buttons were still set to on, just switched to off for next trials. The screenshots from cruiser in the first post were from my second device (Galaxy A52G). But in the past it happened also on both devices for sure with K2. The settings of the auto hide buttons were always on when this happened because this is my default setting.

**Reply by devemux86** [reply]
> There the auto hide buttons were still set to on, just switched to off for next trials.

@DolomitesTom have you made any tests with **disabled** auto hide?
(even at home by switching apps)

I can disable the auto hide **animation**, so immediately visible / invisible.
If that's the cause, we can also make it optional.

**Comment by DolomitesTom** [comment]
> made any tests with disabled auto hide?
> (even at home by switching apps)

Not yet, just trying and let it off for a while for real tests. But snowy roads here, takes a while.

> I can disable the auto hide animation, so immediately visible / invisible.

could be the root cause, let's try, I don't need button switch animation.

**Reply by DolomitesTom** [reply]
No problems till now with auto hide off

**Reply by DolomitesTom** [reply]
Auto hide in general is an important function for me, just animation is waivable I guess.

**Comment by devemux86** [comment]
> No problems till now with auto hide off

Let's then disable the **animations** in auto hide in next version and see how it goes.

@DolomitesTom if you do more testing with auto hide **disabled**, please inform me.

**Comment by devemux86** [comment, marked as answer]
> Let's then disable the animations in auto hide in next version and see how it goes.

Available in [Cruiser 3.0.8](https://play.google.com/store/apps/details?id=gr.talent.cruiser), please see if it works properly now.

**Reply by DolomitesTom** [reply]
Ok, thanks a lot, have switched on the auto hide again. While playing around I couldn't force the problem.
Will see on real tours if this happens again or is fixed.

**Comment by DolomitesTom** [comment]
2 x 150 km tours in the last 2 days. Paused and resumed navigation and put Cruiser to the background several times, also for more than 1,5 hours. No problems any more...

**Reply by devemux86** [reply]
Strange that an animation can cause such a problem, but it's great that we found it.
Thanks again for the detailed report, the tests and help in finding the cause!
